anaconda_project/env_spec.py

```diff
@@ -310,14 +313,22 @@ def inherit_from_names(self):
     def path(self, project_dir):
         """The filesystem path to the default conda env containing our packages."""
         if self._path is None:
-            for base in os.environ.get('ANACONDA_PROJECT_ENVS_PATH', '').split(os.pathsep):
+            env_paths = os.environ.get('ANACONDA_PROJECT_ENVS_PATH', '').split(os.pathsep)
+            for base in reversed(env_paths):
                 base = os.path.expanduser(base) if base else 'envs'
                 path = os.path.abspath(os.path.join(project_dir, base, self.name))
                 found = os.path.isdir(os.path.join(path, 'conda-meta'))
-                if found or self._path is None:
-                    self._path = path
-                    if found:
+                deviations = self._conda.find_environment_deviations(path, self)
```

Comment:
Why should we call `find_environment_deviations` if `found` is False?

Reply:
Yeah, that's not necessary